File: webui/src/services/telegramService.js
import { API_ENDPOINTS } from '@constants/appConstants'

class TelegramService {
    async saveConfig(token, chatId) {
        try {
            const response = await fetch(API_ENDPOINTS.TELEGRAM_CONFIG, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/x-www-form-urlencoded',
                },
                body: `token=${encodeURIComponent(token)}&chat_id=${encodeURIComponent(chatId)}`
            })

            return response.ok
        } catch (error) {
            console.error('Failed to save telegram config:', error)
            return false
        }
    }

    async testConnection() {
        try {
            const response = await fetch(API_ENDPOINTS.TELEGRAM_TEST, {
                method: 'POST'
            })

            return response.ok
        } catch (error) {
            console.error('Failed to test telegram connection:', error)
            return false
        }
    }

    async getStatus() {
        try {
            const response = await fetch(API_ENDPOINTS.TELEGRAM_CONFIG)
            if (!response.ok) throw new Error('Failed to get telegram status')
            return await response.json()
        } catch (error) {
            console.error('Failed to get telegram status:', error)
            return { enabled: false }
        }
    }

    async saveSetting(type, value) {
        try {
            const response = await fetch(API_ENDPOINTS.SETTINGS, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/x-www-form-urlencoded',
                },
                body: `type=${type}&value=${value}`
            })
            return response.ok
        } catch (error) {
            console.error('Failed to save setting:', error)
            return false
        }
    }
}

export const telegramService = new TelegramService()

File: services/telegram_service.go
package services

import (
	"fmt"
	"log"
)

// TelegramService는 텔레그램 관련 비즈니스 로직을 처리합니다
type TelegramService struct {
	container *ServiceContainer
}

// NewTelegramService는 새로운 텔레그램 서비스를 생성합니다
func NewTelegramService(container *ServiceContainer) *TelegramService {
	return &TelegramService{
		container: container,
	}
}

// SaveConfig는 텔레그램 설정을 저장합니다
func (s *TelegramService) SaveConfig(token, chatID string) error {
	err := s.container.Config.SetTelegramConfig(token, chatID)
	if err != nil {
		log.Printf("텔레그램 설정 저장 실패: %v", err)
		return err
	}

	log.Printf("텔레그램 설정 저장됨")
	return nil
}

// GetStatus는 텔레그램 상태를 반환합니다
func (s *TelegramService) GetStatus() map[string]interface{} {
	return map[string]interface{}{
		"enabled": s.container.Config.TelegramEnabled,
	}
}

// TestConnection은 텔레그램 연결을 테스트합니다
func (s *TelegramService) TestConnection() error {
	if !s.container.Config.TelegramEnabled || s.container.Config.TelegramBot == nil {
		return fmt.Errorf("텔레그램이 설정되지 않았습니다")
	}

	err := s.container.Config.TelegramBot.TestConnection()
	if err != nil {
		log.Printf("텔레그램 테스트 실패: %v", err)
		return err
	}

	log.Printf("텔레그램 테스트 성공")
	return nil
}
